### PR TITLE
feat(cardinality): Add cardinality kill switch to global config

### PR DESCRIPTION
**Pull request by Dav1dde**
Sentry PR: https://github.com/getsentry/sentry/pull/63942

#skip-changelog

### CODE REVIEW

File: relay-server/src/services/processor.rs [hunk 1]
```diff
@@ -1811,6 +1811,15 @@ impl EnvelopeProcessorService {
         buckets: Vec<Bucket>,
         mode: ExtractionMode,
     ) -> Vec<Bucket> {
+        use relay_dynamic_config::CardinalityLimiterMode;
```
Comment (olksdr) [inline relay-server/src/services/processor.rs]: we have something like imports only for processing , see 
https://github.com/getsentry/relay/blob/ef5111abf8273b0a3468d26a56867a7e4d0ff645/relay-server/src/services/processor.rs#L44
should this `use`  also be moved in there?

**Review by olksdr — APPROVED** [review]
looks good to me

File: relay-server/src/services/processor.rs (hunk 1 above)
Comment (Dav1dde) [reply relay-server/src/services/processor.rs]: I'll move it 👍

File: relay-dynamic-config/src/global.rs [hunk 2]
```diff
@@ -58,11 +66,29 @@ pub struct Options {
...
 
+/// Kill switch for controlling the cardinality limiter.
+#[derive(Default, Clone, Copy, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub enum CardinalityLimiterMode {
+    /// Cardinality limiter is enabled.
+    #[default]
+    Enabled,
+    /// Cardinality limiter is enabled but no actions are taken.
```
Comment (iker-barriocanal) [inline relay-dynamic-config/src/global.rs]: What does "no actions are taken" mean? Could we elaborate that a bit?

Comment (Dav1dde) [reply relay-dynamic-config/src/global.rs]: Rephrased to:

> Cardinality limiter is enabled but cardinality limits are not enforced.

Is that better to understand?

**Review by iker-barriocanal — APPROVED** [review]

File: relay-dynamic-config/src/global.rs (hunk 2 above)
Comment (iker-barriocanal) [reply relay-dynamic-config/src/global.rs]: Yep, thanks!

**Comment by Dav1dde** [comment]
Don't merge this yet, for some reason I am seeing an empty string being sent as the option, this makes the global config fail to deserialize.

**Comment by Dav1dde** [comment]
We serialize the option from an empty string as well, dealing with the broken values. For the future Sentry will send the correct value in the first place.

File: relay-dynamic-config/src/global.rs [hunk 3]
```diff
@@ -58,11 +66,32 @@ pub struct Options {
...
+    /// Cardinality limiter is enabled.
+    #[default]
+    // De-serialize from the empty string, because the option was added to
+    // Sentry incorrectly which makes Sentry send the empty string as a default.
+    #[serde(alias = "")]
+    Enabled,
+    /// Cardinality limiter is enabled but cardinality limits are not enforced.
+    Passive,
+    /// Cardinality limiter is disabled.
+    Disabled,
```
Comment (jjbayer) [inline relay-dynamic-config/src/global.rs:92]: Is there a chance this enum will be extended in the future? If so, we should add an `Unknown(String)` option, such that external Relays can still parse the config.

**Pull request merged**